client/analytics/components/report-chart/index.js

```diff
@@ -27,17 +29,43 @@ import { Chart, ChartPlaceholder } from 'components';
 import { getReportChartData } from 'store/reports/utils';
 import ReportError from 'analytics/components/report-error';
 
-class ReportChart extends Component {
+export const DEFAULT_FILTER = 'all';
+
+export class ReportChart extends Component {
+	getSelectedFilter( filters, query ) {
+		if ( filters.length === 0 ) {
+			return null;
+		}
+
+		const filterConfig = filters.pop();
+
+		if ( filterConfig.showFilters( query ) ) {
+			const allFilters = flattenFilters( filterConfig.filters );
+			const value = query[ filterConfig.param ] || DEFAULT_FILTER;
+			const selectedFilter = find( allFilters, { value } );
+			const selectedFilterParam = get( selectedFilter, [ 'settings', 'param' ] );
+
+			if ( ! selectedFilterParam || Object.keys( query ).includes( selectedFilterParam ) ) {
+				return selectedFilter;
+			}
+		}
+
+		return this.getSelectedFilter( filters, query );
+	}
+
+	getChartMode() {
+		const { filters, query } = this.props;
+		if ( ! filters ) {
+			return null;
+		}
+		const clonedFilters = filters.slice( 0 );
+		const selectedFilter = this.getSelectedFilter( clonedFilters, query );
+
+		return get( selectedFilter, [ 'chartMode' ] );
```

Comment:
default to `time-comparison` ?

Reply:
I think I prefer passing `null` to the `Chart` component so it can get the value from the `defaultProps`..

This way, if some time we want to change the default display mode of all charts, we will only have to do that in the component `defaultProps`. How does that sound?

In #818 we are going to merge `layout` and `mode` props into one single prop, so the code will be simplified a bit as well.

Reply:
Makes sense to me, thanks.